### ISSUE TITLE
Test 1 fails (JIT wins, classic loses)

### ISSUE THREAD

**Issue by aerkiaga** [aerkiaga/hmars#10]
Warrior that reproduces the bug:

```
;redcode-94
	SPL.B	$16,	#0
	DJN.B	$0,	#100
	ADD.F	}7998,	$7999
	DJN.B	$7999,	#16
	ADD.AB	$7997,	$7997
	MOD.B	$7996,	#2
	JMZ.B	$34,	$7995
	JMP.B	#0,	#0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	JMN.I	>8,	>7992
	MUL.X	*7995,	*0
	MOV.AB	>3,	{7997
	MUL.BA	<7997,	{0
	MUL.I	>1,	$7993
	NOP.A	@1,	{7993
	ADD.F	*5,	>7992
	SEQ.X	{2,	@7998
	SLT.A	<3,	$7997
	SLT.BA	}5,	@8
	SPL.I	$7995,	}7993
	MOV.F	{3,	#7996
	JMP.A	>5,	{7993
	SEQ.A	*4,	@3
	NOP.A	@7,	>7996
	SNE.F	*6,	<7998
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	DAT.F	$0,	$0
	MOV.I	$7968,	$7976
	JMP.B	$7999,	>7999
```

**Comment by aerkiaga** [comment]
Minimal testcase:

```
;redcode-94
ptrb	DAT.F	$0,	$0
start	JMN.I	$ptra,	>ptrb
	JMP.X	#0,	$0
ptra	DAT.F	<0,	$0
	end	start
```

Seems to be another postincrement-related problem.

**Comment by aerkiaga** [comment]
Does not work if the `JMN` at `start` has `.A/.BA` modifier.

**Comment by aerkiaga** [comment]
But it does if the `>` is exchanged with `}`. Apparently the `JMN` gets the postincremented field instead of evaluating it first, then incrementing.

**Comment by aerkiaga** [comment]
Similar to issue #5, fixed by commit 64cff87f2fc9bddc6852250526cdf829745044ac. Now that B-instruction is also buffered, postincrement B-modes must be applied after buffering.
